### ISSUE TITLE
Padded-blocks false positive with decorators

### ISSUE THREAD

**Issue by stremlenye** [eslint/eslint#3097]
version: `v1.0.0-rc-1`
message: `Block must not be padded by blank lines  padded-blocks`

Code:
```js
function foo () {
  @decorator
  class A {

  }
  return A
}
```

Rule:
```json
"parser": "babel-eslint"
…
"padded-blocks": [2, "never"]
```
Version:
```json
"babel-eslint": "^3.1.23"
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by ilyavolodin** [comment]
Which parser are you using?

**Comment by stremlenye** [comment]
```json
"parser": "babel-eslint",
```
```json
"babel-eslint": "^3.1.23"
```

**Comment by ilyavolodin** [comment]
I think this might be `babel-eslint` issue. I do not know what they do with decorators and how they are translated into AST. If they are just removed and empty line is maintained instead, that would be the source of your issue.

**Comment by nzakas** [comment]
I'm not sure there's anything we can do here. Using experimental features means some parts of ESLint aren't going to work.

**Comment by stremlenye** [comment]
Ok. I'll open an issue on babel-eslint.
Thanks for explanation.
Close for now.
